js/components/tooltip.js

```diff
@@ -0,0 +1,135 @@
+var React = require('react');
+var ReactDOM = require('react-dom');
+var deepmerge = require('deepmerge');
+
+var verticalOffset = 80;
+function getContainerStyle(bottom, visible, responsivenessMultiplier, alignment) {
+  var verticalAlignment = verticalOffset * responsivenessMultiplier;
+  var alignmentStyle = {
+    left: {
+      left: 0,
+      transform: 'translate(0,' + verticalAlignment + '%)',
+      '-webkit-transform': 'translate(0,' + verticalAlignment + '%)'
+    },
+    center: {
+      left: '50%',
+      transform: 'translate(-50%,' + verticalAlignment + '%)',
+      '-webkit-transform': 'translate(-50%,' + verticalAlignment + '%)'
+    },
+    right: {
+      right: 0,
+      transform: 'translate(0,' + verticalAlignment + '%)',
+      '-webkit-transform': 'translate(0,' + verticalAlignment + '%)'
+    }
+  };
+
+  var style = {
+    position: 'absolute',
+    color: 'white',
+    fontFamily: '"Roboto Condensed", sans-serif',
+    opacity: visible ? '0.75' : '0',
+    fontWeight: 'normal',
+    bottom: bottom,
+    transition: '1s',
+    visibility: visible ? 'visible' : 'hidden',
+  }
+
+  return deepmerge(alignmentStyle[alignment], style);
+
+}
+
+function getBoxStyle(responsivenessMultiplier) {
+  return {
+    borderRadius: '3px',
+    fontSize: '15px',
+    textShadow: 'none',
+    background: 'black',
+    paddingTop: 8 * responsivenessMultiplier,
+    paddingRight: 20 * responsivenessMultiplier,
+    paddingBottom: 8 * responsivenessMultiplier,
+    paddingLeft: 20 * responsivenessMultiplier
+  }
+}
+
+var pointerAlignment = {
+  left: '10%',
+  center: '45%',
+  right: '85%'
+}
+function getPointerStyle(alignment) {
+  return {
+    position: 'absolute',
+    borderLeft: '5px solid transparent',
+    borderRight: '5px solid transparent',
+    borderTop: '5px solid black',
+    left: pointerAlignment[alignment]
+  };
+}
+
+
+var Tooltip = React.createClass({
+
+  componentDidMount: function () {
+    this.onMouseOver = this.onMouseOver.bind(this);
+    this.onMouseLeave = this.onMouseLeave.bind(this);
+
+    this.parentElement = (ReactDOM.findDOMNode(this) || {}).parentElement;
```

Comment:
Do we need the second half where we get the parentElement of an empty object? If the first part fails to find the DOM node, it should be falsy and the if statement in the next line will fail.